packages/js/api-client/src/client.ts

```diff
@@ -0,0 +1,219 @@
+import { Routes, RoutesMeta } from "./generated/routes"
+
+interface RequestConfig {
+  headers: Headers
+  method: string
+  body?: string | URLSearchParams
+}
+
+export interface OpenverseResponse<Body> {
+  body: Body
+  meta: {
+    headers: Headers
+    status: number
+    url: string
+    request: RequestConfig
+  }
+}
+
+export interface ClientOptions {
+  baseUrl?: string
+  credentials?: {
+    clientId: string
+    clientSecret: string
+  }
+}
+
+/**
+ * Get the timestamp as the number of seconds from the UNIX epoch.
+ * @returns the UNIX timestamp with a resolution of one second
+ */
+const currTimestamp = (): number => Math.floor(Date.now() / 1e3)
+export const expiryThreshold = 5 // seconds
+
+type OpenverseRequest<T extends keyof Routes> = Routes[T]["request"] & {
+  headers?: HeadersInit
+}
+
+interface TransportFetchResponse {
+  headers: Headers
+  status: number
+}
+
+export interface Transport {
+  fetch: (
+    url: string,
+    request: RequestConfig
+  ) => Promise<TransportFetchResponse>
+  getBinaryResponseBody: (res: unknown) => Promise<ArrayBuffer>
+  getJsonResponseBody: (res: unknown) => Promise<unknown>
+}
+
+async function getFetch(): Promise<Transport> {
+  if (!globalThis.fetch) {
+    throw new Error(
+      "Default fetch transport used outside context with global fetch"
+    )
+  }
+
+  return {
+    fetch,
+    getBinaryResponseBody: (res) => (res as Response).arrayBuffer(),
+    getJsonResponseBody: (res) => (res as Response).json(),
+  }
+}
+
+export type OpenverseClient = ReturnType<typeof OpenverseClient>
+export const OpenverseClient = (
+  {
+    baseUrl = "https://api.openverse.engineering/",
```

Comment:
```suggestion
    baseUrl = "https://api.openverse.org/",
```

Shows the age of the original PR :)